File: backend/feature/meditation/src/main/java/com/mindcare/meditation/components/MeditationPlayerDialog.kt
package com.mindcare.meditation.components

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Close
import androidx.compose.material.icons.filled.Pause
import androidx.compose.material.icons.filled.PlayArrow
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.unit.dp
import androidx.hilt.navigation.compose.hiltViewModel
import coil.compose.AsyncImage
import com.mindcare.meditation.MeditationViewModel
import com.mindcare.meditation.model.MeditationSession

@Composable
fun MeditationPlayerDialog(
    session: MeditationSession,
    onDismiss: () -> Unit,
    viewModel: MeditationViewModel = hiltViewModel()
) {
    val isPlaying by viewModel.isPlaying.collectAsState()
    val progress by viewModel.currentProgress.collectAsState()

    Dialog(onDismissRequest = onDismiss) {
        Surface(
            modifier = Modifier
                .fillMaxWidth()
                .wrapContentHeight(),
            shape = RoundedCornerShape(16.dp),
            color = MaterialTheme.colorScheme.surface
        ) {
            Column(
                modifier = Modifier.padding(24.dp),
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Text(
                        text = session.title,
                        style = MaterialTheme.typography.titleLarge
                    )
                    IconButton(onClick = onDismiss) {
                        Icon(Icons.Default.Close, contentDescription = "Cerrar")
                    }
                }
                
                Spacer(modifier = Modifier.height(16.dp))
                
                AsyncImage(
                    model = session.imageUrl,
                    contentDescription = null,
                    modifier = Modifier
                        .size(200.dp)
                        .clip(RoundedCornerShape(8.dp)),
                    contentScale = ContentScale.Crop
                )
                
                Spacer(modifier = Modifier.height(16.dp))
                
                LinearProgressIndicator(
                    progress = progress,
                    modifier = Modifier.fillMaxWidth()
                )
                
                Spacer(modifier = Modifier.height(8.dp))
                
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween
                ) {
                    Text(
                        text = formatDuration((session.duration * progress).toInt()),
                        style = MaterialTheme.typography.bodySmall
                    )
                    Text(
                        text = formatDuration(session.duration),
                        style = MaterialTheme.typography.bodySmall
                    )
                }
                
                Spacer(modifier = Modifier.height(24.dp))
                
                IconButton(
                    onClick = { 
                        if (isPlaying) viewModel.pauseSession() 
                        else viewModel.resumeSession() 
                    },
                    modifier = Modifier.size(64.dp)
                ) {
                    Icon(
                        imageVector = if (isPlaying) Icons.Default.Pause else Icons.Default.PlayArrow,
                        contentDescription = if (isPlaying) "Pausar" else "Reproducir",
                        modifier = Modifier.size(48.dp)
                    )
                }
            }
        }
    }
} 

File: backend/lib/widgets/meditation_player_dialog.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../models/meditation_models.dart';
import '../viewmodels/meditation_viewmodel.dart';
import 'package:flutter/foundation.dart' show kIsWeb;

class MeditationPlayerDialog extends StatelessWidget {
  final MeditationSession session;
  final VoidCallback onDismiss;

  const MeditationPlayerDialog({
    Key? key,
    required this.session,
    required this.onDismiss,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    final viewModel = Provider.of<MeditationViewModel>(context);
    
    return Dialog(
      shape: RoundedRectangleBorder(
        borderRadius: BorderRadius.circular(16.0),
      ),
      child: Padding(
        padding: const EdgeInsets.all(24.0),
        child: Column(
          mainAxisSize: MainAxisSize.min,
          children: [
            // Título y botón de cerrar
            Row(
              mainAxisAlignment: MainAxisAlignment.spaceBetween,
              children: [
                Text(
                  session.title,
                  style: Theme.of(context).textTheme.titleLarge,
                ),
                IconButton(
                  icon: const Icon(Icons.close),
                  onPressed: onDismiss,
                ),
              ],
            ),
            
            const SizedBox(height: 16),
            
            // Imagen
            ClipRRect(
              borderRadius: BorderRadius.circular(8.0),
              child: Image.network(
                session.imageUrl,
                width: 200,
                height: 200,
                fit: BoxFit.cover,
              ),
            ),
            
            const SizedBox(height: 16),
            
            // Mensaje específico para web
            if (kIsWeb && !viewModel.isPlaying && !viewModel.audioInitialized)
              Padding(
                padding: const EdgeInsets.only(bottom: 16.0),
                child: Text(
                  'Haz clic en reproducir para iniciar la meditación',
                  style: TextStyle(
                    color: Theme.of(context).colorScheme.secondary,
                    fontStyle: FontStyle.italic,
                  ),
                  textAlign: TextAlign.center,
                ),
              ),
            
            // Barra de progreso
            LinearProgressIndicator(
              value: viewModel.currentProgress,
              backgroundColor: Theme.of(context).colorScheme.surfaceVariant,
              valueColor: AlwaysStoppedAnimation<Color>(
                Theme.of(context).colorScheme.primary,
              ),
            ),
            
            const SizedBox(height: 8),
            
            // Tiempo
            Row(
              mainAxisAlignment: MainAxisAlignment.spaceBetween,
              children: [
                Text(
                  _formatDuration((session.duration * viewModel.currentProgress).toInt()),
                  style: Theme.of(context).textTheme.bodySmall,
                ),
                Text(
                  _formatDuration(session.duration),
                  style: Theme.of(context).textTheme.bodySmall,
                ),
              ],
            ),
            
            const SizedBox(height: 24),
            
            // Botón de reproducción/pausa
            IconButton(
              iconSize: 48,
              icon: Icon(
                viewModel.isPlaying ? Icons.pause : Icons.play_arrow,
              ),
              onPressed: () {
                if (viewModel.isPlaying) {
                  viewModel.pauseSession();
                } else {
                  viewModel.resumeSession();
                }
              },
            ),
            
            // Mensaje de advertencia para web
            if (kIsWeb)
              Padding(
                padding: const EdgeInsets.only(top: 16.0),
                child: Text(
                  'Nota: Mantén esta pestaña abierta para continuar la reproducción',
                  style: TextStyle(
                    fontSize: 12,
                    color: Theme.of(context).colorScheme.onSurfaceVariant,
                  ),
                  textAlign: TextAlign.center,
                ),
              ),
          ],
        ),
      ),
    );
  }

  String _formatDuration(int durationInSeconds) {
    final minutes = durationInSeconds ~/ 60;
    final seconds = durationInSeconds % 60;
    return '$minutes:${seconds.toString().padLeft(2, '0')}';
  }
} 